### ISSUE TITLE
False positive isinstance-second-argument-not-valid-type when testing typing.Iterable

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#3507]
### Steps to reproduce

```py
from typing import Iterable

print(isinstance([], Iterable))
```

Similar issues exist for other generic collection types such as `Sized`, `Mapping` etc.

### Current behavior

```
Second argument of isinstance is not a type (isinstance-second-argument-not-valid-type)
```

### Expected behavior

No warning is issued.

When `collections.abc.Iterable` is used instead, there is no warning issued. But the way that class is aliased in the `typing` module is probably too complex for static analysis to deal with and there doesn't seem to be explicit support yet to work around that.

### pylint --version output

```
$ pylint --version
pylint 2.5.0-dev1
astroid 2.4.0
Python 3.8.2 (default, Mar 05 2020, 18:58:42) [GCC]
```
The `astroid` version is a dev snapshot as well, but not reported as such. To be exact, I'm using `pylint` at 0ccf1bca and `astroid` at f6a36d91.

**Comment by PCManticore** [comment]
Thanks, I can confirm the bug. It seems we're inferring `typing.Iterable` as an instance rather than a class.

**Comment by bertrand-benoit** [comment]
I confirm the bug with typing.Dict.

**Comment by LuisMarquesSerco** [comment]
Also with typing.List

**Comment by Coffeeri** [comment]
Also with `typing.Generator`.

**Comment by eeroniemi** [comment]
Also happened with `typing.Match`, pylint version 2.5.2.

**Comment by cdce8p** [comment]
This issue will be fixed with the next release of `pylint`. We improved the handling of typing aliases (`Iterable`, `Generator`, `List`, `Match`, and so on) in `astroid` recently.

**Comment by kosyachniy** [comment]
I have the same issue with `typing.Callable`

**Comment by cdce8p** [comment]
> I have the same issue with `typing.Callable`

@kosyachniy Thanks for the report! This will be fixed in the next release of astroid / pylint.
The PR for reference: https://github.com/PyCQA/astroid/pull/1192
